### ISSUE TITLE
Missing dark theme support

### ISSUE THREAD

**Issue by Anth0rx** [portfolio-performance/portfolio#973]
When using a dark theme PP won't adapt to it. It would be cool to support custom colors for icons, fonts diagrams and backgrounds.

---

Current open issues:

- [x] Embedded HTML pie charts should use light/dark CSS
- [x] color of text in diagram legend does not change
- [x] charts still have white background - for example when displayed on the dashboard
- [x] monthly / yearly returns table looses the colored background when in dark mode
- [x] currency diagrams use black color for data series - not visible in dark mode
- [x] Money Trail Tooltip
- [x] Preferences in light mode have grey bars
- [x] highlight colors in tables (for example if the trade is open)
- [ ] check use of ```org.eclipse.swt.display.useSystemTheme```

If you find more issues, post screenshots to this issue.

**Comment by Anth0rx** [comment]
Is there any progress on this matter @buchen?

**Comment by buchen** [comment]
I know Eclipse (the framework I use) has some kind of support for theming. I am happy if somebody can show/explain/prototype how to exactly to this. At the moment, I do not have lot of time (and other feature requests) and unless there are many, many requests, I would not spend time on a dark theme at the moment.

**Comment by Anth0rx** [comment]
Okay, I understand. Unfortunately I do not have enough experience with Java/Eclipse to provide support for this.

**Comment by tomcastro89** [comment]
+1 for this request
The Eclipse IDE itself has a darkmode, but even the "best" there it is not perfect and needs some adjustments ( scrollbars are white etc. )

The strategies are:
1. try to add a dark theme from the eclipse marketplace (https://marketplace.eclipse.org/content/darkest-dark-theme-devstyle)
2. implement a theme ourselves (which takes some effort)

**Comment by buchen** [comment]
So sieht das Dark Theme bei PP aktuell aus:

<img width="1292" alt="Bildschirmfoto 2020-09-13 um 20 27 55" src="https://user-images.githubusercontent.com/587976/93025877-fc820100-f601-11ea-8987-c1b082287c46.png">

**Comment by buchen** [comment]
If you want to help styling the dark theme of PP, then this is a good introductory article:
https://www.vogella.com/tutorials/Eclipse4CSS/article.html

SWT uses CSS files to style the components, but in the end it is a custom implementation, not HTML. Therefore the behavior is not always really predictable. There are lots of custom properties starting with "swt-". With a little bit of trial and error I was able to create the result above.

**Comment by sebasbaumh** [comment]
A little bit of feedback:

The menus still show in gray color though even the Windows 10 setting for apps has been switched to "dark":
![ETTJsJBJpD](https://user-images.githubusercontent.com/670885/93734824-1c558e00-fbdb-11ea-809e-ce34c15561da.png)

Some of the colors look a bit odd (too low contrast on black, here with the selected column in comparison):
![PortfolioPerformance_pSVsjUzq2B](https://user-images.githubusercontent.com/670885/93734848-3db67a00-fbdb-11ea-90ad-4cd937b3fbb1.png)
Also the gray/white border looks a bit too thick, especially in the main view ("Vermögensverwaltung").

The diagram looks ok:
![PortfolioPerformance_aV9afWCtNG](https://user-images.githubusercontent.com/670885/93734885-5cb50c00-fbdb-11ea-96f3-105734388ecf.png)
And one might need to fix colors in the user customizable diagrams to look better on black - but that is expected.

The asset view ("Bestände") is not styled, still white background.

The performance view shows the same exception as in #1751 and here the font for the yearly returns is not ok as well:
![UQI7dK5NM9](https://user-images.githubusercontent.com/670885/93735134-35ab0a00-fbdc-11ea-9588-a6b77ba08bb6.png)
Though the tooltip is fine. It just looks like some of the colors may better be inverted (like black on green instead of white on green).

For the monthly returns maybe the border of the columns could be just a little bit darker:
![PortfolioPerformance_oFi0UlfsZW](https://user-images.githubusercontent.com/670885/93735209-7c006900-fbdc-11ea-8981-e7538baf763f.png)

In "Trades" the text of the yellow column is to light:
![PortfolioPerformance_g3i5AwJU9J](https://user-images.githubusercontent.com/670885/93735251-a6522680-fbdc-11ea-95af-bb607e498d04.png)
It also shows the a bit too heavy grid borders.

The classification views are not styled in dark mode and are light mode like the asset view.

**Comment by buchen** [comment]
@sebasbaumh  thanks! Do you know if Windows has a "native" dark mode itself? At least on macOS, I cannot control the styling of the menu and title bar but need to rely on the native dark mode of the OS itself.


We're getting there...

**Comment by sebasbaumh** [comment]
Yes it has and you can even select to have the taskbar etc. in dark and the apps in light (which looks better by the way 😉 ):
![ApplicationFrameHost_kILCyhL8rB](https://user-images.githubusercontent.com/670885/93735787-79067800-fbde-11ea-903a-46d360b2128e.png)

That's the reason why I tried with both settings set to "dark" - just in case the menus depend on the app setting.
But unfortunately they did not, though that is not a big issue I would say.

**Comment by chrisaut** [comment]
Here's a thread about this:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=540357

It seems they say
"SWT application can now start with the System's theme when the system property org.eclipse.swt.display.useSystemTheme is set to true.
Closing this bug as fixed."

Maybe that's all that is needed?

**Comment by buchen** [comment]
Thanks for the hint about ```org.eclipse.swt.display.useSystemTheme```. The change itself is only about macOS and on macOS there is no problem to dynamically change the theme including menus and title bar. I'll check if the property applies to all. At least it would allow to have three options for the PP theme: automatic, dark, light.

**Comment by chrisaut** [comment]
I think it's more about automatically styling "native" elements, isn't it? Like the very top menu bar that aren't styled with this custom pseudo css. Maybe I understood wrong, I only skimmed through the thread.

**Comment by vwkd** [comment]
Looks like this issue is about Dark Mode which does exists now, except that a "use system theme" option is missing. Would appreciate this to be added.
